### DISCUSSION TITLE
feat: add i18n support

### DISCUSSION THREAD

**Discussion by salihdev0** [acikkaynak/deprem-yardim-frontend#715] [Ideas]
## Define your request
i18n support should be added so that other citizens can use our site.

-- fill in the line below:
** discord username: @ss.ibrahimbas#4990 **

## Define the alternatives you have in mind

I can quickly add this with the [i18next package](https://github.com/i18next/next-i18next)

## Extra context
A reference from discord by @eraygundogmus: https://discord.com/channels/1072074800622739476/1073669869825962027/1073673398439383214

**Comment by myagizmaktav** [comment]
We can use the translatepackage https://www.npmjs.com/package/translatte for this issue instead of adding languages by hand.

Then next step is we can translate this marker tweets info.

![image](https://user-images.githubusercontent.com/115627863/218185658-7bc3cafc-46ed-4431-a023-8759c54d05de.png)

@ssibrahimbas

**Comment by berkevaroll** [comment]
I am taking this issue to work on with @tlgync. I think having tweets translated has more priority since it will give the insights to foreign users

**Comment by usirin** [comment]
What will be the strategy @berkevaroll is it 18next? or translatte?

**Comment by berkevaroll** [comment]
@usirin I checked translatte first to sort it out quickly but it seems that they do not have typescript definitions. And with i18n it would be too complex to define locales as we can't know what they could type. Checking other solutions that can help

**Comment by berkevaroll** [comment]
@myagizmaktav @ssibrahimbas @eraygundogmus  Recommended package for translating tweets does not work anymore since Google and other Translate API providers are no more free. If we want to auto translate them we need some Google (or any other provider) account to use paid API keys.

**Comment by myagizmaktav** [comment]
@berkevaroll  I think translate @type/definitions is not important right now. and translatte is free.
You can use any type.

**Comment by berkevaroll** [comment]
@myagizmaktav I was not mentioning the types, it also uses google api key and etc. That package also depends on Google Translate free api and Google does not provide free service anymore. 
https://console.cloud.google.com/tos?id=translate&pli=1#:~:text=The%20Google%20Translate%20API%20is,com%2Fcloud%2Fpricing%2F.

**Comment by myagizmaktav** [comment]
@usirin  can solve this paid issue I think.

**Comment by 0xCakin** [comment]
Can't we solve this with OpenAI library? We already have their API Key?

**Comment by berkevaroll** [comment]
@canerakin111 I think it only works for English to Spanish,Japanese and French. https://platform.openai.com/examples/default-translate

**Comment by 0xCakin** [comment]
Also what's the point? The priority is the ER teams. They will be reading the tweet so they can find the people. What value do we get if an English speaker reads the tweet? Chrome browser has a default English translate option.

**Comment by berkevaroll** [comment]
@canerakin111 it had p0 tag, and it was said to prioritize them. You are right with that point
